src/Extension/FluentExtension.php

```diff
@@ -835,4 +846,121 @@ public function cacheKeyComponent()
     {
         return 'fluentlocale-' . FluentState::singleton()->getLocale();
     }
+
+    /**
+     * @param FieldList $fields
+     */
+    public function updateCMSFields(FieldList $fields)
+    {
+        // get all fields to translate and remove
+        $translated = $this->getLocalisedTables();
+
+        $currentLocale = Locale::getCurrentLocale();
+        $locale = $currentLocale->Locale;
+        $language = $currentLocale ? strtok($locale, '_') : '';
+
+        foreach ($translated as $table => $translatedFields) {
+            foreach ($translatedFields as $translatedField) {
+                // Find field matching this translated field
+                // If the translated field has an ID suffix also check for the non-suffixed version
+                // E.g. UploadField()
+                $field = $fields->dataFieldByName($translatedField);
+                if (!$field && preg_match('/^(?<field>\w+)ID$/', $translatedField, $matches)) {
+                    $field = $fields->dataFieldByName($matches['field']);
+                }
+
+                // Highlight any translatable field
+                if ($field && !$field->hasClass('LocalisedField')) {
```

Comment:
you can reduce nesting by inverting this.


```php
if (!$field || $field->hasClass('LocalisedField')) {
	continue;
}
```

Reply:
If I were to make this change then the hasClass() function may be called on null.

Reply:
If `!$field` evaluates to true, which it would if `$field` is null, then you wouldn't have that happen